clients/src/test/java/org/apache/kafka/common/serialization/SerializationTest.java

```diff
@@ -24,34 +24,53 @@
 
 public class SerializationTest {
 
-    private static class SerDeser<T> {
-        final Serializer<T> serializer;
-        final Deserializer<T> deserializer;
+    final private String topic = "testTopic";
 
-        public SerDeser(Serializer<T> serializer, Deserializer<T> deserializer) {
-            this.serializer = serializer;
-            this.deserializer = deserializer;
-        }
+    private class DummyClass {
+
+    }
+
+    @Test
+    public void testSerdeFrom() {
+        Serde<Long> thisSerde = Serdes.serdeFrom(Long.class);
+        Serde<Long> otherSerde = Serdes.Long();
+
+        Long value = 423412424L;
+
+        assertEquals("Should get the original long after serialization and deserialization",
+                value, thisSerde.deserializer().deserialize(topic, otherSerde.serializer().serialize(topic, value)));
+        assertEquals("Should get the original long after serialization and deserialization",
+                value, otherSerde.deserializer().deserialize(topic, thisSerde.serializer().serialize(topic, value)));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testSerdeFromUnknown() {
+        Serdes.serdeFrom(DummyClass.class);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testSerdeFromNotNull() {
+        Serdes.serdeFrom(null, Serdes.Long().deserializer());
     }
 
     @Test
     public void testStringSerializer() {
         String str = "my string";
-        String mytopic = "testTopic";
+
         List<String> encodings = new ArrayList<String>();
         encodings.add("UTF8");
         encodings.add("UTF-16");
 
         for (String encoding : encodings) {
-            SerDeser<String> serDeser = getStringSerDeser(encoding);
-            Serializer<String> serializer = serDeser.serializer;
-            Deserializer<String> deserializer = serDeser.deserializer;
+            Serde<String> serDeser = getStringSerDeser(encoding);
+            Serializer<String> serializer = serDeser.serializer();
+            Deserializer<String> deserializer = serDeser.deserializer();
 
             assertEquals("Should get the original string after serialization and deserialization with encoding " + encoding,
-                    str, deserializer.deserialize(mytopic, serializer.serialize(mytopic, str)));
+                    str, deserializer.deserialize(topic, serializer.serialize(topic, str)));
```

Comment:
Because for some customized serde, their behavior would be different given the topic.